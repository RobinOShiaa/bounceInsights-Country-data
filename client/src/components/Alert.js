import { CheckCircleIcon } from "@heroicons/react/20/solid";

export const Alert = ({ value }) => {
  const primaryColour = value ? "text-green-400" : "text-red-400";
  const ringColour = value ? "ring-green-600" : "ring-red-600";
  const ringOffsetColour = value
    ? "ring-offset-green-50"
    : "ring-offset-red-50";
  const secondaryColour = value ? "text-green-800" : "text-red-800";
  const bgColour = value ? "bg-green-50" : "bg-red-50";

  return (
    <div className="rounded-md bg-green-50 p-4">
      <div className="flex">
        <div className="flex-shrink-0">
          <CheckCircleIcon
            className={`h-5 w-5  ${primaryColour}`}
            aria-hidden="true"
          />
        </div>
        <div className="ml-3">
          <p
            data-testid="result"
            className={`text-sm font-medium ${secondaryColour}`}
          >
            {value
              ? "Country data retrieved"
              : "Country data retrieval unsuccessful"}
          </p>
        </div>
        <div className="ml-auto pl-3">
          <div className="-mx-1.5 -my-1.5">
            <button
              type="button"
              className={`inline-flex rounded-md ${bgColour} p-1.5 ${primaryColour} hover:bg-green-100 focus:outline-none focus:ring-2 focus:${ringColour} focus:ring-offset-2 focus:${ringOffsetColour}`}
            ></button>
          </div>
        </div>
      </div>
    </div>
  );
};
