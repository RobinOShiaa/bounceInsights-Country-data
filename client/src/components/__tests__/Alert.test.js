import { Alert } from "../Alert";
import { render } from "@testing-library/react";


test("Matches Alert Succesfull component", () => {
    const tree = render(<Alert value={true}/>);
    expect(tree.getByTestId('result').textContent).toEqual('Country data retrieved')
});

test("Matches Alert Unsuccesfull component", () => {
    const tree = render(<Alert value={false}/>);
    expect(tree.getByTestId('result').textContent).toEqual('Country data retrieval unsuccessful')
});



test("Matches Alert Succesfull snapshot", () => {
    const tree = render(<Alert value={true}/>);
    expect(tree).toMatchSnapshot();
});

test("Matches Alert Unsuccesfull snapshot", () => {
    const tree = render(<Alert value={false}/>);
    expect(tree).toMatchSnapshot();
});
  
  