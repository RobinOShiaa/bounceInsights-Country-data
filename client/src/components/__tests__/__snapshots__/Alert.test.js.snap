// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Alert Succesfull snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-md bg-green-50 p-4"
      >
        <div
          class="flex"
        >
          <div
            class="flex-shrink-0"
          >
            <svg
              aria-hidden="true"
              class="h-5 w-5  text-green-400"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="ml-3"
          >
            <p
              class="text-sm font-medium text-green-800"
              data-testid="result"
            >
              Country data retrieved
            </p>
          </div>
          <div
            class="ml-auto pl-3"
          >
            <div
              class="-mx-1.5 -my-1.5"
            >
              <button
                class="inline-flex rounded-md bg-green-50 p-1.5 text-green-400 hover:bg-green-100 focus:outline-none focus:ring-2 focus:ring-green-600 focus:ring-offset-2 focus:ring-offset-green-50"
                type="button"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-md bg-green-50 p-4"
    >
      <div
        class="flex"
      >
        <div
          class="flex-shrink-0"
        >
          <svg
            aria-hidden="true"
            class="h-5 w-5  text-green-400"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="ml-3"
        >
          <p
            class="text-sm font-medium text-green-800"
            data-testid="result"
          >
            Country data retrieved
          </p>
        </div>
        <div
          class="ml-auto pl-3"
        >
          <div
            class="-mx-1.5 -my-1.5"
          >
            <button
              class="inline-flex rounded-md bg-green-50 p-1.5 text-green-400 hover:bg-green-100 focus:outline-none focus:ring-2 focus:ring-green-600 focus:ring-offset-2 focus:ring-offset-green-50"
              type="button"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Matches Alert Unsuccesfull snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-md bg-green-50 p-4"
      >
        <div
          class="flex"
        >
          <div
            class="flex-shrink-0"
          >
            <svg
              aria-hidden="true"
              class="h-5 w-5  text-red-400"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="ml-3"
          >
            <p
              class="text-sm font-medium text-red-800"
              data-testid="result"
            >
              Country data retrieval unsuccessful
            </p>
          </div>
          <div
            class="ml-auto pl-3"
          >
            <div
              class="-mx-1.5 -my-1.5"
            >
              <button
                class="inline-flex rounded-md bg-red-50 p-1.5 text-red-400 hover:bg-green-100 focus:outline-none focus:ring-2 focus:ring-red-600 focus:ring-offset-2 focus:ring-offset-red-50"
                type="button"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-md bg-green-50 p-4"
    >
      <div
        class="flex"
      >
        <div
          class="flex-shrink-0"
        >
          <svg
            aria-hidden="true"
            class="h-5 w-5  text-red-400"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="ml-3"
        >
          <p
            class="text-sm font-medium text-red-800"
            data-testid="result"
          >
            Country data retrieval unsuccessful
          </p>
        </div>
        <div
          class="ml-auto pl-3"
        >
          <div
            class="-mx-1.5 -my-1.5"
          >
            <button
              class="inline-flex rounded-md bg-red-50 p-1.5 text-red-400 hover:bg-green-100 focus:outline-none focus:ring-2 focus:ring-red-600 focus:ring-offset-2 focus:ring-offset-red-50"
              type="button"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
